### PR TITLE
feat: add support for PT1.12.1

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*Issue #, if available:*

*Description of changes:*

add support for PT1.12.1

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by tejaschumbalkar** [comment]
Please let me know if there are other places where the patch version needs to be updated

File: src/sagemaker/fw_utils.py [hunk 1]
```diff
@@ -122,7 +122,7 @@
         "1.11",
         "1.11.0",
         "1.12",
-        "1.12.0",
+        "1.12.1",
```
Comment (saimidu) [inline src/sagemaker/fw_utils.py:125]: `1.12.1` should be an addition, not a replacement.

File: src/sagemaker/fw_utils.py [hunk 2]
```diff
@@ -133,7 +133,7 @@
     "1.11",
     "1.11.0",
     "1.12",
-    "1.12.0",
+    "1.12.1",
```
Comment (saimidu) [inline src/sagemaker/fw_utils.py:136]: `1.12.1` should be an addition, not a replacement.

File: tests/unit/test_fw_utils.py [hunk 3]
```diff
@@ -863,7 +863,7 @@ def test_validate_smdataparallel_args_not_raises():
         ("ml.p3.16xlarge", "pytorch", "1.10", "py38", smdataparallel_enabled),
         ("ml.p3.16xlarge", "pytorch", "1.11.0", "py38", smdataparallel_enabled),
         ("ml.p3.16xlarge", "pytorch", "1.11", "py38", smdataparallel_enabled),
-        ("ml.p3.16xlarge", "pytorch", "1.12.0", "py38", smdataparallel_enabled),
+        ("ml.p3.16xlarge", "pytorch", "1.12.1", "py38", smdataparallel_enabled),
```
Comment (saimidu) [inline tests/unit/test_fw_utils.py:866]: `1.12.1` should be an addition, not a replacement.

File: tests/unit/test_fw_utils.py [hunk 4]
```diff
@@ -881,7 +881,7 @@ def test_validate_smdataparallel_args_not_raises():
         ("ml.p3.16xlarge", "pytorch", "1.9.1", "py38", smdataparallel_enabled_custom_mpi),
         ("ml.p3.16xlarge", "pytorch", "1.10.2", "py38", smdataparallel_enabled_custom_mpi),
         ("ml.p3.16xlarge", "pytorch", "1.11.0", "py38", smdataparallel_enabled_custom_mpi),
-        ("ml.p3.16xlarge", "pytorch", "1.12.0", "py38", smdataparallel_enabled_custom_mpi),
+        ("ml.p3.16xlarge", "pytorch", "1.12.1", "py38", smdataparallel_enabled_custom_mpi),
```
Comment (saimidu) [inline tests/unit/test_fw_utils.py:884]: `1.12.1` should be an addition, not a replacement.

File: tests/unit/test_fw_utils.py [hunk 5]
```diff
@@ -916,7 +916,7 @@ def test_validate_pytorchddp_not_raises():
         "1.11",
         "1.11.0",
         "1.12",
-        "1.12.0",
+        "1.12.1",
```
Comment (saimidu) [inline tests/unit/test_fw_utils.py:919]: `1.12.1` should be an addition, not a replacement.

File: src/sagemaker/image_uri_config/pytorch.json [hunk 6]
```diff
@@ -650,6 +650,38 @@
                     "us-west-1": "763104351884",
                     "us-west-2": "763104351884"
                 },
+            "1.12.1": {
```
Comment (claytonparnell) [inline src/sagemaker/image_uri_config/pytorch.json:653]: It looks like you accidentally put the 1.12.1 block inside the 1.12.0 block

**Review by claytonparnell — COMMENTED** [review]
/bot run slow-tests

**Review by saimidu — APPROVED** [review]

**Comment by saimidu** [comment]
@claytonparnell Could you please take another look at the failing slow/notebook tests? They seem completely unrelated to the changes made in this PR, as per the CI Reports.

**Review by mufaddal-rohawala — COMMENTED** [review]
/bot run slow-tests, notebook-tests

**Review by mufaddal-rohawala — APPROVED** [review]

**Pull request merged**